packages/arb-token-bridge-ui/src/hooks/useDeposits.ts

```diff
@@ -12,68 +12,117 @@ import {
 import { useNetworksAndSigners } from './useNetworksAndSigners'
 import { Transaction } from './useTransactions'
 import {
-  getQueryParamsForFetchingReceivedFunds,
-  getQueryParamsForFetchingSentFunds
+  SubgraphQueryTypes,
+  SubgraphQueryTypeCount,
+  countSubgraphQueryTypes,
+  getAdditionalSubgraphQueryParams
 } from '../util/SubgraphUtils'
+import { useIsConnectedToArbitrum } from './useIsConnectedToArbitrum'
+import { useAccountType } from './useAccountType'
 
 export type CompleteDepositData = {
   deposits: Transaction[]
   pendingDeposits: Transaction[]
   transformedDeposits: MergedTransaction[]
 }
 
-export const fetchCompleteDepositData = async (
+export const fetchCompleteDepositData = async ({
+  walletAddress,
+  depositQueryTypes,
+  depositsQueryCount,
+  depositParams
+}: {
+  walletAddress: string
+  depositQueryTypes: Partial<SubgraphQueryTypes>[]
+  depositsQueryCount: SubgraphQueryTypeCount
   depositParams: FetchDepositParams
-): Promise<CompleteDepositData> => {
+}): Promise<CompleteDepositData> => {
+  // create queries for each SubgraphQueryType for deposits
+  // we will fetch them all, and in the next steps we decide which of them to display
+  const promises = depositQueryTypes.map(type =>
+    fetchDeposits({
+      ...depositParams,
+      ...getAdditionalSubgraphQueryParams(type, walletAddress),
+      subgraphQueryType: type,
```

Comment:
The reason why `subgraphQueryType` is passed for deposits and not withdrawals is the need for calling different endpoints for deposits (`/deposits` or `/retryables`). 

We don't utilize it in this PR but I've already made it support this, as it will be needed for the custom destination ETH history.